src/device/pf_fspm.c

```diff
@@ -365,6 +474,14 @@ void pf_fspm_create_log_book_entry (
       net->fspm_log_book.entries[put].pnio_status = *p_pnio_status;
       net->fspm_log_book.entries[put].entry_detail = entry_detail;
       os_mutex_unlock (net->fspm_log_book_mutex);
+
+      LOG_DEBUG (
+         PNET_LOG,
+         "FSPM(%d): Added logbook entry to position %u\n",
+         __LINE__,
+         put);
+
+      pf_fspm_save_logbook_if_modified (net);
```

Comment:
It seems like the create function is intended to be used from different contexts since log_book data is protected by the mutex.  However the log_book is not protected during save operation. which may lead to that a faulty log_book is written. Maybe problem can solved by using a temp buffer .  Extending locked block to include save operation is not a good idea

Reply:
Now using temporary buffers for both read and write.
Done

Reply:
ok